polaris/hub/polarisfs.py

```diff
@@ -212,3 +239,30 @@ def pipe_file(
             timeout=timeout,
         )
         response.raise_for_status()
+
+        if self.use_listings_cache:
+            new_listing = [{"name": path, "type": "file", "size": len(content)}]
+            current_listings = self.dircache.get(self._parent(path), None)
+            if current_listings is not None:
+                new_listing.extend(current_listings)
+                self.dircache[self._parent(path)] = new_listing
+            else:
+                self.dircache[self._parent(path)] = new_listing
+
+    def info(self, path, **kwargs):
+        path = self._strip_protocol(path)
+        out = self.ls(self._parent(path), detail=True, **kwargs)
+        out = [o for o in out if o["name"].rstrip("/") == path]
+        if out:
+            return out[0]
+        out = self.ls(path, detail=True, **kwargs)
+        path = path.rstrip("/")
+        out1 = [o for o in out if o["name"].rstrip("/") == path]
+        if len(out1) == 1:
+            if "size" not in out1[0]:
+                out1[0]["size"] = None
+            return out1[0]
+        elif len(out1) > 1:
+            return {"name": path, "size": 0, "type": "directory"}
+        else:
+            raise FileNotFoundError(path)
```

Comment:
This is exactly the same as it is in the parent class, right? No need to overwrite it?

Reply:
Yes, I had it slightly changed (originally `elif len(out1) > 1 or out`) as it was resulting in an error, but appears to be good without it now! :+1: